### ISSUE TITLE
Track Name Inconsistency When Derived From Search (feat., remix)

### ISSUE THREAD

**Issue by carlyd95** [zotify-dev/zotify#111]
It seems for some reason adding (feat. ?) or (with ?)  or [feat. ] or other changes related to remixes are appended to the track title.

Examples:
(LINK)
(API TRACK NAME) ==> (ZOTIFY FILENAME USED)

[Wildfire - Mielo Remix by Mielo](https://open.spotify.com/track/1UBOzykJgq5r7bsxFx04Ys)
Wildfire - Mielo Remix ==> Slow Magic - Wildfire (Mielo Remix).{EXT}

[Move All Night by SNBRN](https://open.spotify.com/track/6RKFXPdoCBRcLljsxKZUNw)
Move All Night ==> Move All Night (feat. Kole).{EXT}

This even occurs if you use a custom output path. It seems the root issue lies in {song_name} as it is manipulated beyond what the Spotify api presents the track title as.

Could this behavior please be made a flag that we can turn it off with?

As a background I am creating .m3u8's so it would be nice to have consistently named files.

I will probably create a fork of this this weekend and see if I can modify it to accommodate this change.

Under closer investigation a track's name may vary depending on directly accessing the track via the api versus using the search feature.

When downloading a playlist is Zotify currently searching each and every track in the playlist via the search feature as opposed to directly accessing the track via its track_id?

Any thoughts and/or suggestions are welcome.

**Demonstration:**
`import spotipy
from spotipy.oauth2 import SpotifyOAuth

# Your Spotify API credentials
CLIENT_ID = 'beep'
CLIENT_SECRET = 'boop'
REDIRECT_URI = 'http://localhost:8888/callback'  # You need to specify this in your Spotify developer dashboard

# Initialize Spotipy with OAuth authentication
sp = spotipy.Spotify(auth_manager=SpotifyOAuth(client_id=CLIENT_ID,
                                               client_secret=CLIENT_SECRET,
                                               redirect_uri=REDIRECT_URI,
                                               scope='playlist-read-private'))

def get_artist_and_track_name(track_uri):
    # Get track details
    track_info = sp.track(track_uri)
    track_name = track_info['name']
    artist_name = track_info['artists'][0]['name']  # Assuming we only consider the first artist
    return artist_name, track_name

def search_track(artist_name, track_name):
    # Search for the track
    search_results = sp.search(q=f'artist:{artist_name} track:{track_name}', type='track', limit=1)
    if search_results['tracks']['items']:
        return search_results['tracks']['items'][0]['name']
    else:
        return None

# Playlist ID
playlist_id = '0uXbw4hGH88fKH3oljrYX1'

# Get tracks in the playlist
results = sp.playlist_items(playlist_id, fields='items(track(uri))')

# Iterate over each track in the playlist
for item in results['items']:
    track_uri = item['track']['uri']
    artist_name, track_name = get_artist_and_track_name(track_uri)
    print("Playlist Track Name:", track_name)
    
    # Search for the track and print the name of the first result
    search_result = search_track(artist_name, track_name)
    if search_result:
        print("First Search Result Track Name:", search_result)
    else:
        print("No search result found.")`

**This script yields:**

Playlist Track Name: Move All Night
First Search Result Track Name: Move All Night (feat. Kole)

**Comment by carlyd95** [comment]
So TL;DR: the track name used in a playlist can vary from the track name used in a track specific call by track ID.

Kinda weird but that is how Spotify has the data set up...

For anyone else interested I did the following:

in track.py changed line 145: `def download_track(mode: str, track_id: str, extra_keys=None, disable_progressbar=False) -> None:` to `def download_track(mode: str, track_id: str, track_name_override=None, extra_keys=None, disable_progressbar=False) -> None:` 

and added `if track_name_override != None:
            name = track_name_override` after lines 157-159 `(artists, raw_artists, album_name, name, image_url, release_year, disc_number,
         track_number, scraped_song_id, is_playable, duration_ms) = get_song_info(track_ID)`

I am closing this issue.
